cmd/orb-server/startcmd/params.go

```diff
@@ -1517,9 +1534,27 @@ func getOrbParameters(cmd *cobra.Command) (*orbParameters, error) {
 		currentSidetreeProtocolVersion:          currentSidetreeProtocolVersion,
 		kmsParams:                               kmsParams,
 		requestTokens:                           requestTokens,
+		metricsProviderName:                     metricsProviderName,
+		prometheusMetricsProviderParams:         prometheusMetricsProviderParams,
 	}, nil
 }
 
+func getMetricsProviderName(cmd *cobra.Command) (string, error) {
+	metricsProvider, err := cmdutil.GetUserSetVarFromString(cmd, metricsProviderFlagName, metricsProviderEnvKey, true)
+	if err != nil {
+		return "", err
+	}
+	return metricsProvider, nil
+}
+
+func getPrometheusMetricsProviderParams(cmd *cobra.Command) (*prometheusMetricsProviderParams, error) {
+	promMetricsUrl, err := cmdutil.GetUserSetVarFromString(cmd, promHttpUrlFlagName, promHttpUrlEnvKey, false)
+	if err != nil {
+		return nil, err
```

Comment:
You should provide context when returning an error. For example:

return nil, fmt.Errorf("%s: %w", promHttpUrlFlagName, err)

Reply:
actually the err already contains that context. When I tried to add the context as per your suggestion, it was mentioning the flag name twice in the error message, so for now, I have kept this as it is, lets discuss if you still think this needs to be modified.